tests/defer/src/commonTest/kotlin/test/DeferTest.kt

```diff
@@ -0,0 +1,271 @@
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", null),
+                )
+            )
+        )
+            .errors(
+                listOf(
+                    Error(
+                        message = "Cannot resolve isColor",
+                        locations = listOf(Error.Location(1, 119)),
+                        path = listOf("computers", 0, "screen", "isColor"),
+                        extensions = null, nonStandardFields = null
+                    )
+                )
+            )
+            .build(),
+
+        ApolloResponse.Builder(
+            query,
+            uuid,
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", ComputerFields("486", 1996,
+                        ComputerFields.Screen("Screen", "800x600", null))),
+                )
+            )
+        ).build(),
+
+        ApolloResponse.Builder(
+            query,
+            uuid,
+            data = WithFragmentSpreadsQuery.Data(
+                listOf(
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer1", ComputerFields("386", 1993,
+                        ComputerFields.Screen("Screen", "640x480", null))),
+                    WithFragmentSpreadsQuery.Computer("Computer", "Computer2", ComputerFields("486", 1996,
+                        ComputerFields.Screen("Screen", "800x600",
+                            ScreenFields(true)))),
+                )
+            )
+        ).build(),
+    )
+
+    mockServer.enqueueMultipart(jsonList)
+    val actualResponseList = apolloClient.query(query).toFlow().toList()
+    assertResponseListEquals(expectedDataList, actualResponseList)
+  }
+}
+
+private fun assertResponseListEquals(expectedDataList: List<ApolloResponse<*>>, actualResponseList: List<ApolloResponse<*>>) {
+  assertContentEquals(expectedDataList, actualResponseList) { expectedResponse, actualResponse ->
+    assertEquals(expectedResponse.data, actualResponse.data)
+    assertContentEquals(expectedResponse.errors, actualResponse.errors) { expectedError, actualError ->
+      assertEquals(expectedError.message, actualError.message)
+      assertContentEquals(expectedError.path, actualError.path)
+      assertContentEquals(expectedError.locations, actualError.locations) { expectedLocation, actualLocation ->
+        assertEquals(expectedLocation.line, actualLocation.line)
+        assertEquals(expectedLocation.column, actualLocation.column)
```

Comment:
No data class for Errors 😅. Not sure if there's a specific reason for that. Let's keep this in mind, we could add it if we find more evidence this could help.

Reply:
Yeah wasn't sure if it was OK to change it at this point :) We could also only add equals/hashCode without making a data class if that's a bit less "API changing".